File: E-bank/src/main/java/com/ebank/model/Account.java
package com.ebank.model;

import com.ebank.enums.AccountType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import jakarta.persistence.*;

import java.sql.Date;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Account {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String accountNumber;
    @Enumerated(EnumType.STRING)
    private AccountType accountType;
    private double balance;
    private Date dateCreation;
    private Boolean accountClosed = false;
    private String raisonClosing;

    @ManyToOne
    @JsonIgnore
    private User user;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Transaction> transactions = new ArrayList<>();;

    @OneToMany(mappedBy = "account", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<BankCard> bankCards = new ArrayList<>();;

    @OneToMany(cascade = CascadeType.ALL)
    private List<Beneficiary> beneficiaries = new ArrayList<>();;

}

File: Ebank-Frontend/src/app/model/account/account.ts
import { Transaction } from '../transaction/transaction';
import { BankCard } from '../bank-card/bankcard';
import { Beneficiary } from '../beneficiary/beneficiary';
import {AccountType} from "../../enum/account-type";


export class Account {
  id: number;
  accountNumber: string;
  accountType: AccountType;
  balance: number;
  dateCreation: Date;
  accountClosed: boolean;
  raisonClosing: string;
  userId: number;
  transactions: Transaction[];
  bankCards: BankCard[];
  beneficiaries: Beneficiary[];

  constructor(
    id: number,
    accountNumber: string,
    accountType: AccountType,
    balance: number,
    dateCreation: Date,
    accountClosed: boolean,
    raisonClosing: string,
    userId: number,
    transactions: Transaction[] = [],
    bankCards: BankCard[] = [],
    beneficiaries: Beneficiary[] = []
  ) {
    this.id = id;
    this.accountNumber = accountNumber;
    this.accountType = accountType;
    this.balance = balance;
    this.dateCreation = dateCreation;
    this.accountClosed = accountClosed;
    this.raisonClosing = raisonClosing;
    this.userId = userId;
    this.transactions = transactions;
    this.bankCards = bankCards;
    this.beneficiaries = beneficiaries;
  }
}
